vt build with openmpi fails

/Users/fnrizzi/Desktop/work/darma/VT/testBuild_clang80/vt/vt/lib/fmt/fmt/core.h:895:3: error: static_assert failed due to requirement 'std::is_same<char, ompi_op_t>::value'
      "mismatch between char-types of context and argument"
  static_assert(std::is_same<typename C::char_type,
  ^             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/Users/fnrizzi/Desktop/work/darma/VT/testBuild_clang80/vt/vt/lib/fmt/fmt/core.h:1083:10: note: in instantiation of function template specialization
      'fmt::v5::internal::make_value<fmt::v5::basic_format_context<std::__1::back_insert_iterator<fmt::v5::internal::buffer<char> >, char>, ompi_op_t *, 0>' requested here
  return make_value<Context>(value);
         ^
/Users/fnrizzi/Desktop/work/darma/VT/testBuild_clang80/vt/vt/lib/fmt/fmt/core.h:1199:25: note: in instantiation of function template specialization 'fmt::v5::internal::make_arg<true,
      fmt::v5::basic_format_context<std::__1::back_insert_iterator<fmt::v5::internal::buffer<char> >, char>, ompi_op_t *, 0>' requested here
      : data_{internal::make_arg<IS_PACKED, Context>(args)...} {}
                        ^
/Users/fnrizzi/Desktop/work/darma/VT/testBuild_clang80/vt/vt/lib/fmt/fmt/core.h:1377:10: note: in instantiation of member function
      'fmt::v5::format_arg_store<fmt::v5::basic_format_context<std::__1::back_insert_iterator<fmt::v5::internal::buffer<char> >, char>, void *, unsigned long, const char *, short, int, unsigned long, const
      char *, ompi_op_t *>::format_arg_store' requested here
  return {args...};
         ^
/Users/fnrizzi/Desktop/work/darma/VT/testBuild_clang80/vt/vt/lib/fmt/fmt/core.h:1467:39: note: in instantiation of function template specialization 'fmt::v5::internal::make_args_checked<char [57], void *,
      unsigned long, const char *, short, int, unsigned long, const char *, ompi_op_t *, 0>' requested here
                           {internal::make_args_checked(format_str, args...)});
                                      ^
/Users/fnrizzi/Desktop/work/darma/VT/testBuild_clang80/vt/vt/src/vt/configs/debug/debug_print.h:172:22: note: in instantiation of function template specialization 'fmt::v5::format<char [57], void *, unsigned
      long, const char *, short, int, unsigned long, const char *, ompi_op_t *, 0>' requested here
    auto user = fmt::format(std::forward<Arg>(arg),std::forward<Args>(args)...);
                     ^
/Users/fnrizzi/Desktop/work/darma/VT/testBuild_clang80/vt/vt/src/vt/configs/debug/debug_print.h:193:7: note: (skipping 18 contexts in backtrace; use -ftemplate-backtrace-limit=0 to see all)
      debugPrintImpl<cat,mod>(node,std::forward<Arg>(arg),std::forward<Args>(args)...);
      ^
/opt/local/include/gtest/internal/gtest-internal.h:721:57: note: in instantiation of member function 'testing::internal::TypeParameterizedTest<TestRDMAHandleSet,
      testing::internal::TemplateSel<test_rdma_handle_set_3>, testing::internal::Types8<double, float, int, long long, unsigned long long, long long, short, unsigned short> >::Register' requested here
                                 typename Types::Tail>::Register(prefix,
                                                        ^
/opt/local/include/gtest/internal/gtest-internal.h:770:50: note: in instantiation of member function 'testing::internal::TypeParameterizedTest<TestRDMAHandleSet,
      testing::internal::TemplateSel<test_rdma_handle_set_3>, testing::internal::Types9<int, double, float, int, long long, unsigned long long, long long, short, unsigned short> >::Register' requested here
    TypeParameterizedTest<Fixture, Head, Types>::Register(
                                                 ^
/opt/local/include/gtest/internal/gtest-internal.h:775:47: note: in instantiation of member function 'testing::internal::TypeParameterizedTestSuite<TestRDMAHandleSet,
      testing::internal::Templates3<test_rdma_handle_set_3, test_rdma_handle_set_4, test_rdma_handle_set_5>, testing::internal::Types9<int, double, float, int, long long, unsigned long long, long long,
      short, unsigned short> >::Register' requested here
                                      Types>::Register(prefix, code_location,
                                              ^
/opt/local/include/gtest/internal/gtest-internal.h:775:47: note: in instantiation of member function 'testing::internal::TypeParameterizedTestSuite<TestRDMAHandleSet,
      testing::internal::Templates4<test_rdma_handle_set_2, test_rdma_handle_set_3, test_rdma_handle_set_4, test_rdma_handle_set_5>, testing::internal::Types9<int, double, float, int, long long, unsigned
      long long, long long, short, unsigned short> >::Register' requested here
/Users/fnrizzi/Desktop/work/darma/VT/testBuild_clang80/vt/vt/tests/unit/rdma/test_rdma_static_sub_handle.cc:389:1: note: in instantiation of member function
      'testing::internal::TypeParameterizedTestSuite<TestRDMAHandleSet, testing::internal::Templates5<test_rdma_handle_set_1, test_rdma_handle_set_2, test_rdma_handle_set_3, test_rdma_handle_set_4,
      test_rdma_handle_set_5>, testing::internal::Types9<int, double, float, int, long long, unsigned long long, long long, short, unsigned short> >::Register' requested here
INSTANTIATE_TYPED_TEST_SUITE_P(
^
/opt/local/include/gtest/gtest-typed-test.h:311:11: note: expanded from macro 'INSTANTIATE_TYPED_TEST_SUITE_P'
          Register(#Prefix,                                                 \
          ^
In file included from /Users/fnrizzi/Desktop/work/darma/VT/testBuild_clang80/vt/vt/tests/unit/rdma/test_rdma_static_sub_handle.cc:47:
In file included from /Users/fnrizzi/Desktop/work/darma/VT/testBuild_clang80/vt/vt/src/vt/transport.h:48:
In file included from /Users/fnrizzi/Desktop/work/darma/VT/testBuild_clang80/vt/vt/src/vt/config.h:48:
In file included from /Users/fnrizzi/Desktop/work/darma/VT/testBuild_clang80/vt/vt/src/vt/configs/debug/debug_masterconfig.h:59:
In file included from /Users/fnrizzi/Desktop/work/darma/VT/testBuild_clang80/vt/vt/src/vt/configs/debug/debug_print.h:53:
In file included from /Users/fnrizzi/Desktop/work/darma/VT/testBuild_clang80/vt/vt/lib/fmt/fmt/format.h:60:
/Users/fnrizzi/Desktop/work/darma/VT/testBuild_clang80/vt/vt/lib/fmt/fmt/core.h:898:10: error: no viable conversion from returned value of type 'basic_string_view<ompi_op_t>' to function return type
      'init<fmt::v5::basic_format_context<std::__1::back_insert_iterator<fmt::v5::internal::buffer<char> >, char>, basic_string_view<typename basic_format_context<back_insert_iterator<buffer<char> >,
      char>::char_type>, string_type>' (aka 'init<fmt::v5::basic_format_context<std::__1::back_insert_iterator<fmt::v5::internal::buffer<char> >, char>, basic_string_view<char>, string_type>')
  return to_string_view(val);
         ^~~~~~~~~~~~~~~~~~~
/Users/fnrizzi/Desktop/work/darma/VT/testBuild_clang80/vt/vt/lib/fmt/fmt/core.h:759:59: note: candidate constructor (the implicit copy constructor) not viable: no known conversion from
      'basic_string_view<ompi_op_t>' to 'const fmt::v5::internal::init<fmt::v5::basic_format_context<std::__1::back_insert_iterator<fmt::v5::internal::buffer<char> >, char>, fmt::v5::basic_string_view<char>,
      fmt::v5::internal::string_type> &' for 1st argument
template <typename Context, typename T, type TYPE> struct init {
                                                          ^
/Users/fnrizzi/Desktop/work/darma/VT/testBuild_clang80/vt/vt/lib/fmt/fmt/core.h:759:59: note: candidate constructor (the implicit move constructor) not viable: no known conversion from
      'basic_string_view<ompi_op_t>' to 'fmt::v5::internal::init<fmt::v5::basic_format_context<std::__1::back_insert_iterator<fmt::v5::internal::buffer<char> >, char>, fmt::v5::basic_string_view<char>,
      fmt::v5::internal::string_type> &&' for 1st argument
template <typename Context, typename T, type TYPE> struct init {
                                                          ^
/Users/fnrizzi/Desktop/work/darma/VT/testBuild_clang80/vt/vt/lib/fmt/fmt/core.h:763:17: note: candidate constructor not viable: no known conversion from 'basic_string_view<ompi_op_t>' to
      'const fmt::v5::basic_string_view<char> &' for 1st argument
  FMT_CONSTEXPR init(const T& v) : val(v) {}
                ^
In file included from /Users/fnrizzi/Desktop/work/darma/VT/testBuild_clang80/vt/vt/tests/unit/rdma/test_rdma_static_sub_handle.cc:45:
In file included from /opt/local/include/gtest/gtest.h:58:
In file included from /opt/local/libexec/llvm-8.0/bin/../include/c++/v1/ostream:138:
In file included from /opt/local/libexec/llvm-8.0/bin/../include/c++/v1/ios:216:
In file included from /opt/local/libexec/llvm-8.0/bin/../include/c++/v1/__locale:15:
In file included from /opt/local/libexec/llvm-8.0/bin/../include/c++/v1/string:505:
In file included from /opt/local/libexec/llvm-8.0/bin/../include/c++/v1/string_view:176:
/opt/local/libexec/llvm-8.0/bin/../include/c++/v1/__string:86:55: error: variable has incomplete type 'std::__1::char_traits<ompi_op_t>::char_type' (aka 'ompi_op_t')
    static inline _LIBCPP_CONSTEXPR bool eq(char_type __c1, char_type __c2) _NOEXCEPT
                                                      ^
/opt/local/libexec/llvm-8.0/bin/../include/c++/v1/__string:135:13: note: in instantiation of member function 'std::__1::char_traits<ompi_op_t>::eq' requested here
    for (; !eq(*__s, char_type(0)); ++__s)
            ^
/Users/fnrizzi/Desktop/work/darma/VT/testBuild_clang80/vt/vt/lib/fmt/fmt/core.h:409:49: note: in instantiation of member function 'std::__1::char_traits<ompi_op_t>::length' requested here
      : data_(s), size_(std::char_traits<Char>::length(s)) {}
                                                ^
/Users/fnrizzi/Desktop/work/darma/VT/testBuild_clang80/vt/vt/lib/fmt/fmt/core.h:504:10: note: in instantiation of member function 'fmt::v5::basic_string_view<ompi_op_t>::basic_string_view' requested here
  return s;
         ^
/Users/fnrizzi/Desktop/work/darma/VT/testBuild_clang80/vt/vt/lib/fmt/fmt/core.h:898:10: note: in instantiation of function template specialization 'fmt::v5::to_string_view<ompi_op_t>' requested here
  return to_string_view(val);
         ^
/Users/fnrizzi/Desktop/work/darma/VT/testBuild_clang80/vt/vt/lib/fmt/fmt/core.h:1199:25: note: in instantiation of function template specialization 'fmt::v5::internal::make_arg<true,
      fmt::v5::basic_format_context<std::__1::back_insert_iterator<fmt::v5::internal::buffer<char> >, char>, ompi_op_t *, 0>' requested here
      : data_{internal::make_arg<IS_PACKED, Context>(args)...} {}
                        ^
/Users/fnrizzi/Desktop/work/darma/VT/testBuild_clang80/vt/vt/lib/fmt/fmt/core.h:1377:10: note: (skipping 21 contexts in backtrace; use -ftemplate-backtrace-limit=0 to see all)
  return {args...};
         ^
/opt/local/include/gtest/internal/gtest-internal.h:721:57: note: in instantiation of member function 'testing::internal::TypeParameterizedTest<TestRDMAHandleSet,
      testing::internal::TemplateSel<test_rdma_handle_set_3>, testing::internal::Types8<double, float, int, long long, unsigned long long, long long, short, unsigned short> >::Register' requested here
                                 typename Types::Tail>::Register(prefix,
                                                        ^
/opt/local/include/gtest/internal/gtest-internal.h:770:50: note: in instantiation of member function 'testing::internal::TypeParameterizedTest<TestRDMAHandleSet,
      testing::internal::TemplateSel<test_rdma_handle_set_3>, testing::internal::Types9<int, double, float, int, long long, unsigned long long, long long, short, unsigned short> >::Register' requested here
    TypeParameterizedTest<Fixture, Head, Types>::Register(
                                                 ^
/opt/local/include/gtest/internal/gtest-internal.h:775:47: note: in instantiation of member function 'testing::internal::TypeParameterizedTestSuite<TestRDMAHandleSet,
      testing::internal::Templates3<test_rdma_handle_set_3, test_rdma_handle_set_4, test_rdma_handle_set_5>, testing::internal::Types9<int, double, float, int, long long, unsigned long long, long long,
      short, unsigned short> >::Register' requested here
                                      Types>::Register(prefix, code_location,
                                              ^
/opt/local/include/gtest/internal/gtest-internal.h:775:47: note: in instantiation of member function 'testing::internal::TypeParameterizedTestSuite<TestRDMAHandleSet,
      testing::internal::Templates4<test_rdma_handle_set_2, test_rdma_handle_set_3, test_rdma_handle_set_4, test_rdma_handle_set_5>, testing::internal::Types9<int, double, float, int, long long, unsigned
      long long, long long, short, unsigned short> >::Register' requested here
/Users/fnrizzi/Desktop/work/darma/VT/testBuild_clang80/vt/vt/tests/unit/rdma/test_rdma_static_sub_handle.cc:389:1: note: in instantiation of member function
      'testing::internal::TypeParameterizedTestSuite<TestRDMAHandleSet, testing::internal::Templates5<test_rdma_handle_set_1, test_rdma_handle_set_2, test_rdma_handle_set_3, test_rdma_handle_set_4,
      test_rdma_handle_set_5>, testing::internal::Types9<int, double, float, int, long long, unsigned long long, long long, short, unsigned short> >::Register' requested here
INSTANTIATE_TYPED_TEST_SUITE_P(
^
/opt/local/include/gtest/gtest-typed-test.h:311:11: note: expanded from macro 'INSTANTIATE_TYPED_TEST_SUITE_P'
          Register(#Prefix,                                                 \
          ^
/Users/fnrizzi/tpls/openmpi/4.0.2/install-clang8.0/include/mpi.h:406:16: note: forward declaration of 'ompi_op_t'
typedef struct ompi_op_t *MPI_Op;
               ^
In file included from /Users/fnrizzi/Desktop/work/darma/VT/testBuild_clang80/vt/vt/tests/unit/rdma/test_rdma_static_sub_handle.cc:45:
In file included from /opt/local/include/gtest/gtest.h:58:
In file included from /opt/local/libexec/llvm-8.0/bin/../include/c++/v1/ostream:138:
In file included from /opt/local/libexec/llvm-8.0/bin/../include/c++/v1/ios:216:
In file included from /opt/local/libexec/llvm-8.0/bin/../include/c++/v1/__locale:15:
In file included from /opt/local/libexec/llvm-8.0/bin/../include/c++/v1/string:505:
In file included from /opt/local/libexec/llvm-8.0/bin/../include/c++/v1/string_view:176:
/opt/local/libexec/llvm-8.0/bin/../include/c++/v1/__string:86:71: error: variable has incomplete type 'std::__1::char_traits<ompi_op_t>::char_type' (aka 'ompi_op_t')
    static inline _LIBCPP_CONSTEXPR bool eq(char_type __c1, char_type __c2) _NOEXCEPT
                                                                      ^
/Users/fnrizzi/tpls/openmpi/4.0.2/install-clang8.0/include/mpi.h:406:16: note: forward declaration of 'ompi_op_t'
typedef struct ompi_op_t *MPI_Op;
               ^
In file included from /Users/fnrizzi/Desktop/work/darma/VT/testBuild_clang80/vt/vt/tests/unit/rdma/test_rdma_static_sub_handle.cc:45:
In file included from /opt/local/include/gtest/gtest.h:58:
In file included from /opt/local/libexec/llvm-8.0/bin/../include/c++/v1/ostream:138:
In file included from /opt/local/libexec/llvm-8.0/bin/../include/c++/v1/ios:216:
In file included from /opt/local/libexec/llvm-8.0/bin/../include/c++/v1/__locale:15:
In file included from /opt/local/libexec/llvm-8.0/bin/../include/c++/v1/string:505:
In file included from /opt/local/libexec/llvm-8.0/bin/../include/c++/v1/string_view:176:
/opt/local/libexec/llvm-8.0/bin/../include/c++/v1/__string:135:22: error: 'std::__1::char_traits<ompi_op_t>::char_type' (aka 'ompi_op_t') is an incomplete type
    for (; !eq(*__s, char_type(0)); ++__s)
                     ^
/Users/fnrizzi/Desktop/work/darma/VT/testBuild_clang80/vt/vt/lib/fmt/fmt/core.h:409:49: note: in instantiation of member function 'std::__1::char_traits<ompi_op_t>::length' requested here
      : data_(s), size_(std::char_traits<Char>::length(s)) {}
                                                ^
/Users/fnrizzi/Desktop/work/darma/VT/testBuild_clang80/vt/vt/lib/fmt/fmt/core.h:504:10: note: in instantiation of member function 'fmt::v5::basic_string_view<ompi_op_t>::basic_string_view' requested here
  return s;
         ^
/Users/fnrizzi/Desktop/work/darma/VT/testBuild_clang80/vt/vt/lib/fmt/fmt/core.h:898:10: note: in instantiation of function template specialization 'fmt::v5::to_string_view<ompi_op_t>' requested here
  return to_string_view(val);
         ^
/Users/fnrizzi/Desktop/work/darma/VT/testBuild_clang80/vt/vt/lib/fmt/fmt/core.h:1199:25: note: in instantiation of function template specialization 'fmt::v5::internal::make_arg<true,
      fmt::v5::basic_format_context<std::__1::back_insert_iterator<fmt::v5::internal::buffer<char> >, char>, ompi_op_t *, 0>' requested here
      : data_{internal::make_arg<IS_PACKED, Context>(args)...} {}
                        ^
/Users/fnrizzi/Desktop/work/darma/VT/testBuild_clang80/vt/vt/lib/fmt/fmt/core.h:1377:10: note: in instantiation of member function
      'fmt::v5::format_arg_store<fmt::v5::basic_format_context<std::__1::back_insert_iterator<fmt::v5::internal::buffer<char> >, char>, void *, unsigned long, const char *, short, int, unsigned long, const
      char *, ompi_op_t *>::format_arg_store' requested here
  return {args...};
         ^
/Users/fnrizzi/Desktop/work/darma/VT/testBuild_clang80/vt/vt/lib/fmt/fmt/core.h:1467:39: note: (skipping 20 contexts in backtrace; use -ftemplate-backtrace-limit=0 to see all)
                           {internal::make_args_checked(format_str, args...)});
                                      ^
/opt/local/include/gtest/internal/gtest-internal.h:721:57: note: in instantiation of member function 'testing::internal::TypeParameterizedTest<TestRDMAHandleSet,
      testing::internal::TemplateSel<test_rdma_handle_set_3>, testing::internal::Types8<double, float, int, long long, unsigned long long, long long, short, unsigned short> >::Register' requested here
                                 typename Types::Tail>::Register(prefix,
                                                        ^
/opt/local/include/gtest/internal/gtest-internal.h:770:50: note: in instantiation of member function 'testing::internal::TypeParameterizedTest<TestRDMAHandleSet,
      testing::internal::TemplateSel<test_rdma_handle_set_3>, testing::internal::Types9<int, double, float, int, long long, unsigned long long, long long, short, unsigned short> >::Register' requested here
    TypeParameterizedTest<Fixture, Head, Types>::Register(
                                                 ^
/opt/local/include/gtest/internal/gtest-internal.h:775:47: note: in instantiation of member function 'testing::internal::TypeParameterizedTestSuite<TestRDMAHandleSet,
      testing::internal::Templates3<test_rdma_handle_set_3, test_rdma_handle_set_4, test_rdma_handle_set_5>, testing::internal::Types9<int, double, float, int, long long, unsigned long long, long long,
      short, unsigned short> >::Register' requested here
                                      Types>::Register(prefix, code_location,
                                              ^
/opt/local/include/gtest/internal/gtest-internal.h:775:47: note: in instantiation of member function 'testing::internal::TypeParameterizedTestSuite<TestRDMAHandleSet,
      testing::internal::Templates4<test_rdma_handle_set_2, test_rdma_handle_set_3, test_rdma_handle_set_4, test_rdma_handle_set_5>, testing::internal::Types9<int, double, float, int, long long, unsigned
      long long, long long, short, unsigned short> >::Register' requested here
/Users/fnrizzi/Desktop/work/darma/VT/testBuild_clang80/vt/vt/tests/unit/rdma/test_rdma_static_sub_handle.cc:389:1: note: in instantiation of member function
      'testing::internal::TypeParameterizedTestSuite<TestRDMAHandleSet, testing::internal::Templates5<test_rdma_handle_set_1, test_rdma_handle_set_2, test_rdma_handle_set_3, test_rdma_handle_set_4,
      test_rdma_handle_set_5>, testing::internal::Types9<int, double, float, int, long long, unsigned long long, long long, short, unsigned short> >::Register' requested here
INSTANTIATE_TYPED_TEST_SUITE_P(
^
/opt/local/include/gtest/gtest-typed-test.h:311:11: note: expanded from macro 'INSTANTIATE_TYPED_TEST_SUITE_P'
          Register(#Prefix,                                                 \
          ^
/Users/fnrizzi/tpls/openmpi/4.0.2/install-clang8.0/include/mpi.h:406:16: note: forward declaration of 'ompi_op_t'
typedef struct ompi_op_t *MPI_Op;
               ^
5 errors generated.
make[2]: *** [tests/CMakeFiles/test_rdma_static_sub_handle.dir/unit/rdma/test_rdma_static_sub_handle.cc.o] Error 1
make[1]: *** [tests/CMakeFiles/test_rdma_static_sub_handle.dir/all] Error 2
make: *** [all] Error 2

--- a/src/vt/rdmahandle/holder.impl.h
+++ b/src/vt/rdmahandle/holder.impl.h
@@ -221,8 +221,8 @@ T Holder<T,E>::fetchOp(vt::NodeType node, Lock l, T in, int offset, MPI_Op op) {
     LockMPI _scope_lock(l, node, data_window_);
     debug_print_verbose(
       rdma, node,
-      "MPI_Fetch_and_op({}, {}, {}, {}, {}, {}, window);\n",
-      in, print_ptr(&out), mpi_type_str, node, offset, op
+      "MPI_Fetch_and_op({}, {}, {}, {}, {}, window);\n",
+      in, print_ptr(&out), mpi_type_str, node, offset
     );
     MPI_Fetch_and_op(&in, &out, mpi_type, node, offset, op, data_window_);
   }
@@ -242,8 +242,8 @@ RequestHolder Holder<T,E>::raccum(
       LockMPI _scope_lock(l, node, data_window_);
       debug_print_verbose(
         rdma, node,
-        "MPI_Accumulate({}, {}, {}, {}, {}, {}, {}, {}, window);\n",
-        print_ptr(ptr), len, mpi_type_str, node, offset, len, mpi_type_str, op
+        "MPI_Accumulate({}, {}, {}, {}, {}, {}, {}, window);\n",
+        print_ptr(ptr), len, mpi_type_str, node, offset, len, mpi_type_str
       );
       MPI_Accumulate(
         ptr, len, mpi_type, node, offset, len, mpi_type, op, data_window_
@@ -253,8 +253,8 @@ RequestHolder Holder<T,E>::raccum(
     LockMPI _scope_lock(l, node, data_window_);
     debug_print_verbose(
       rdma, node,
-      "MPI_Raccumulate({}, {}, {}, {}, {}, {}, {}, {}, window);\n",
-      print_ptr(ptr), len, mpi_type_str, node, offset, len, mpi_type_str, op
+      "MPI_Raccumulate({}, {}, {}, {}, {}, {}, {}, window);\n",
+      print_ptr(ptr), len, mpi_type_str, node, offset, len, mpi_type_str
     );
     MPI_Raccumulate(
       ptr, len, mpi_type, node, offset, len, mpi_type, op, data_window_, r.add()

